src/client.js

```diff
@@ -19,81 +19,147 @@ class TouchPortalClient extends EventEmitter {
     this.customStates = {};
   }
 
-  createState(id, desc, defaultValue, parentGroup) {
-    if (this.customStates[id]) {
-      this.logIt('ERROR', `createState: Custom state of ${id} already created`);
-      throw new Error(`createState: Custom state of ${id} already created`);
-    }
-    this.customStates[id] = desc;
-    const newState = {
-      type: 'createState',
-      id: `${id}`,
-      desc: `${desc}`,
-      defaultValue: `${defaultValue}`,
-    };
-    if (parentGroup !== '' || parentGroup !== undefined) {
-      newState.parentGroup = `${parentGroup}`;
-    }
-    this.send(newState);
-  }
-
-  createStateMany(states) {
-    const createStateArray = [];
-
-    if (states.length <= 0) {
-      this.logIt('ERROR', 'createStateMany : states contains no data');
-      throw new Error('createStateMany: states contains no data');
+  /**
+   * Internal helper method for states
+   * 
+   * @param {Object[Object]} states - Array of object containing id,
+   * desc, defaultValue, optionally parentGroup.
+   * @param {String} stateType - TouchPortal state type removeState
+   * or createState.
+   * 
+   * @return {Array<Object>}
+   */
+  stateHelper(states, stateType) {
+    const stateArray = [];
+    if (!(typeof states == "object" || states.length)) {
+      this.logIt('ERROR', `${stateType} has to be vaild object.`)
     }
-
     states.forEach((state) => {
-      if (this.customStates[state.id]) {
-        this.logIt('WARN', `createState: Custom state of ${state.id} already created`);
-      } else {
-        this.customStates[state.id] = state.desc;
-        const newState = {
-          type: 'createState',
-          id: `${state.id}`,
-          desc: `${state.desc}`,
-          defaultValue: `${state.defaultValue}`,
+      const stateObject = {
+        type: `${stateType}`,
+        id: `${state.id}`
+      };
+      if (`${stateType}` === "createState") {
+        if (this.customStates[state.id]) {
+          this.logIt('ERROR', `createState: Custom state of ${state.id} already created`);
+          throw new Error(`createState: Custom state of ${state.id} already created`);
+        };
+        stateObject.desc = state.desc;
+        stateObject.defaultValue = state.defaultValue
+        if (state.parentGroup) {
+          stateObject.parentGroup = `${state.parentGroup}`;
         };
```

Comment:
This should also add the id to `this.customStates`

```
this.customStates[state.id] = state.desc;
```

Reply:
Thanks!